docs/src/main/asciidoc/kubernetes-client.adoc

```diff
@@ -0,0 +1,79 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Kubernetes Client
+
+include::./attributes.adoc[]
+
+
+Quarkus includes the `kubernetes-client` extension which enables the use of the [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client)
+in native mode while also making it easier to work with.
+
+== Configuration
+
+Once you have your Quarkus project configured you can add the `kubernetes-client` extension
+to your project by running the following from the command line in your project base directory.
+
+[source]
+mvn quarkus:add-extension -Dextensions="kubernetes-client"
+
+This will add the following to your pom.xml
+
+[source,xml]
+----
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-kubernetes-client</artifactId>
+    </dependency>
+----
+
+== Testing
+
+To make testing against a mock Kubernetes API extremely simple, Quarkus provides the `KubernetesMockServerTestResource` which automatically launches
+a mock of the Kubernetes API server and sets the proper environment variables needed so that the Kubernetes Client configures itself to use said mock. +
+Tests can inject the mock and set it up in any way necessary for the particular testing using the `@MockServer` annotation.
+
+An example of such a test could look like the following:
```

Comment:
`Here is an example of such test:`
maybe?

Reply:
Sounds better! Fixed